collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -349,12 +349,6 @@ func (s *Stream) read(_ context.Context) error {
 		// timeout.  TODO: possibly, improve to wait for no outstanding requests and then stop reading.
 		resp, err := s.client.Recv()
 		if err != nil {
-			// Once the send direction of stream is closed the server should return
-			// an error that mentions an EOF. The expected error code is codes.Unknown.
-			status, ok := status.FromError(err)
-			if ok && status.Message() == "EOF" && status.Code() == codes.Unknown {
-				return nil
-			}
```

Comment:
What happens now, when `nil` is returned. How does the read call return?

Reply:
Thanks for the review! It seems returning nil was not enough to prevent the client from seeing an error, so I am now calling Send from the server before returning nil.

So now:
1. client calls CloseSend() after `max_stream_lifetime` elapses
2. server receives EOF from client to indicate stream closing
3. server sends response with `batchID=-1` and `status=OK` to client. Then server returns nil.
4. client receives response with `batchID=-1` and `status=OK` at which point `stream.read` will return
5. stream is restarted